### PR TITLE
[TDF] Implement  Cache in terms of TLazyDS (ROOT-9226)

### PR DESCRIPTION
**Pull request by dpiparo**
And simplify the code whenever possible.

### CODE REVIEW

File: tree/treeplayer/test/dataframe/dataframe_cache.cxx [hunk 1]
```diff
@@ -31,6 +31,8 @@ TEST(Cache, FundType)
    }
 }
 
+/*
+// The contiguity has been dropped since now caching is backed up by a TDS
```
Comment (bluehood) [inline tree/treeplayer/test/dataframe/dataframe_cache.cxx]: i'd add something like `// FIXME readd test when TLazyDS does not make internal copies anymore`

File: tree/treeplayer/inc/ROOT/TDFInterface.hxx [hunk 2]
```diff
@@ -728,7 +729,7 @@ public:
    /// This action is *lazy*: upon invocation of this method the calculation is
    /// booked but not executed. See TResultPtr documentation.
    template <typename T, typename COLL = std::vector<T>>
-   TResultPtr<typename TDFDetail::ColType_t<T, COLL>> Take(std::string_view column = "")
+   TResultPtr<COLL> Take(std::string_view column = "")
```
Comment (bluehood) [inline tree/treeplayer/inc/ROOT/TDFInterface.hxx]: iiuc now `Take<TVec<T>>` returns a `TResultPtr<TVec<T>>` instead of a `TResultPtr<std::vector<T>>` like before.
this change in return type probably deserves its own commit.

lifetimes are ok right? we will never return adopting `TVec`s I guess..?

Comment (dpiparo) [reply tree/treeplayer/inc/ROOT/TDFInterface.hxx]: I thought about this. We may be so efficient (0 copies) that we may end up with an adopting TVec. I'll force the realloc explicitly.
